app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -267,10 +260,67 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         })
     }
 
+    fun observeForToolbar(fragment: Fragment) {
+        // TODO remove this
+        // See comment in fun observeForMegaTile
+        val factory = fragment.context!!.serviceLocator.viewModelFactory
+        toolbarViewModel = ViewModelProviders.of(fragment, factory).get(ToolbarViewModel::class.java)
+
+        fun updateOverlayButtonState(isEnabled: Boolean, overlayButton: ImageButton) {
+            overlayButton.isEnabled = isEnabled
+            overlayButton.isFocusable = isEnabled
+            overlayButton.alpha =
```

Comment:
(Follow-up?) Should the alpha be defined as part of the "enabled/disabled" state instead?

Reply:
That does seem cleaner, but I'm also wary of hiding styles behind more cascading XML files.  I could go either way on this one.